src/typography/body2/style.tsx

```diff
@@ -0,0 +1,25 @@
+import styled from 'styled-components';
+import { colors } from '../../colors/colors';
+import { typography } from '../typography';
+import { Body2Props } from '.';
+
+type Colors = 'cta' | 'gray' | 'grayDark' | 'errorDark';
+
+const color = {
+  base: {
+    cta: colors.brand.ctaDark,
+    gray: colors.base.gray,
+    grayDark: colors.base.grayDark,
+    errorDark: colors.state.statesError,
+  },
+};
+
+export const StyledBody2 = styled.p<Body2Props>`
+  font-family: ${typography.family.primary};
+  font-size: ${(props) => (props.size === 'large' ? typography.fontSize.large : typography.fontSize.small)};
```

Comment:
Dá pra criar uma interface e adicionar o tipo de `props` aqui pra não precisar usar o `props.color as Colors`
```suggestion
  font-size: ${(props: StyledBody2Props) => (props.size === 'large' ? typography.fontSize.large : typography.fontSize.small)};
```